core/drivers/gic.c

```diff
@@ -190,17 +191,22 @@ void gic_init(struct gic_data *gd, vaddr_t gicc_base __maybe_unused,
 	 */
 #if defined(CFG_ARM_GICV3)
 	write_icc_pmr(0x80);
-	write_icc_ctlr(GICC_CTLR_ENABLEGRP0 | GICC_CTLR_ENABLEGRP1 |
-		       GICC_CTLR_FIQEN);
+	write_icc_igrpen1(1);
+
+	write32(read32(gd->gicd_base + GICD_CTLR) | GICD_CTLR_ENABLEGRP0 |
+		GICD_CTLR_ENABLEGRP1 | GICD_CTLR_ENABLEGRP1S,
```

Comment:
Please do not enable Group0 and Group1NS as they are meant to be initialized by EL3 and non-secure world respectively.

Reply:
Agree.  GICv3 should only enable G1S.  GICv2 case can enable G0 only.
Thanks I will make this change.